samples/apply-dictionary-renderer-to-graphics-overlay/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertographicsoverlay/components/ApplyDictionaryRendererToGraphicsOverlayViewModel.kt

```diff
@@ -0,0 +1,252 @@
+     */
+    private suspend fun createMil2525dDictionaryRenderer(): DictionaryRenderer? {
+        val portalItem =
+            PortalItem(Portal.arcGISOnline(Portal.Connection.Anonymous), "d815f3bdf6e6452bb8fd153b654c94ca")
+
+        val dictionarySymbolStyle = DictionarySymbolStyle(portalItem = portalItem)
+
+        dictionarySymbolStyle.load().onFailure { throwable ->
+            messageDialogVM.showMessageDialog(throwable)
+            return null
+        }
+
+        // Uses the "Ordered Anchor Points" for the symbol style draw rule.
+        dictionarySymbolStyle.configurations.firstOrNull { it.name.equals("model", ignoreCase = true) }
+            ?.let { configuration -> configuration.value = "ORDERED ANCHOR POINTS" }
+
+        return DictionaryRenderer(dictionarySymbolStyle = dictionarySymbolStyle)
+    }
+
+    /**
+     * Create point graphics from a local XML file containing Mil2525d message data.
+     * The XML file is downloaded by the downloader activity.
+     */
+    private fun makeMessageGraphics(): List<Graphic> {
+        val xmlFile = File(provisionPath, "mil2525dmessages.xml")
+        val messageXml = xmlFile.readText()
+        val parser = MessageXmlParser()
+        val messages = parser.parse(messageXml)
+
+        return messages.mapNotNull { message ->
+            val wkid = message.wkid
+            val controlPoints = message.controlPoints
+            if (wkid == null || controlPoints.isEmpty()) {
+                // Optionally log or handle the error here
+                null
+            } else {
+                try {
+                    val spatialReference = SpatialReference(wkid = wkid)
+                    val points = controlPoints.map { (x, y) ->
+                        Point(x = x, y = y, spatialReference = spatialReference)
+                    }
+                    Graphic(geometry = Multipoint(points), attributes = message.other)
+                } catch (ex: Exception) {
+                    messageDialogVM.showMessageDialog(
+                        title = "Unable to create graphic from XML file.", description = ex.message.toString()
+                    )
+                    null
+                }
+            }
+        }
+    }
+}
+
+private const val TAG_MESSAGES = "messages"
+private const val TAG_MESSAGE = "message"
+private const val TAG_CONTROL_POINTS = "_control_points"
+private const val TAG_WKID = "_wkid"
+
+/**
+ * Simple XML parser for the Mil2525d message XML file.
+ * This is a basic implementation and does not cover all edge cases.
+ */
+class MessageXmlParser {
+
+    /**
+     * Parses the provided XML string and returns a list of Message objects.
+     */
+    fun parse(xml: String): List<Message> {
+        val messages = mutableListOf<Message>()
+
+        val factory = XmlPullParserFactory.newInstance()
+        val parser = factory.newPullParser()
+        parser.setInput(xml.reader())
```

Comment:
Looking at the Android [docs](https://developer.android.com/develop/connectivity/network-ops/xml?hl=en#instantiate), I would recommend to avoid `newInstance`: 

```suggestion
        val parser = Xml.newPullParser().apply { setInput(xml.reader()) }
```